dataframely/_storage/delta.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) QuantCo 2025-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+from collections.abc import Iterable
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import polars as pl
+
+from ._base import (
+    SerializedCollection,
+    SerializedRules,
+    SerializedSchema,
+    StorageBackend,
+)
+from ._exc import assert_metadata
+from .constants import COLLECTION_METADATA_KEY, RULE_METADATA_KEY, SCHEMA_METADATA_KEY
+
+if TYPE_CHECKING:
+    import deltalake
+
+
+class DeltaStorageBackend(StorageBackend):
+    def sink_frame(
+        self, lf: pl.LazyFrame, serialized_schema: SerializedSchema, **kwargs: Any
+    ) -> None:
+        _raise_on_lazy_write()
```

Comment:
Why the indirection with the separate method?

Reply:
I raise the same exception a bunch of times (also on `sink_collection`, `sink_failure_info`), so I decided to factor it out

Reply:
Ah, I missed this, sorry; fine then!